src/Compilers/CSharp/Test/Semantic/Semantics/RefFieldTests.cs

```diff
@@ -9448,19 +8895,63 @@ class Program
 {{
     static void F(S s)
     {{
-        {refModifier} scoped S s1 = ref s;
         scoped {refModifier} S s2 = ref s;
+    }}
+}}";
+            var comp = CreateCompilation(source);
+            comp.VerifyDiagnostics();
+        }
+
+        [Theory]
+        [InlineData("ref         ")]
+        [InlineData("ref readonly")]
+        public void ScopedRefAndRefStructOnly_05_RefScoped(string refModifier)
+        {
+            var source =
+$@"struct S {{ }}
+class Program
+{{
+    static void F(S s)
+    {{
+        {refModifier} scoped S s1 = ref s;
         scoped {refModifier} scoped S s3 = ref s;
     }}
 }}";
             var comp = CreateCompilation(source);
             comp.VerifyDiagnostics(
-                // (6,29): error CS9048: The 'scoped' modifier can be used for refs and ref struct values only.
+                // (6,22): error CS0246: The type or namespace name 'scoped' could not be found (are you missing a using directive or an assembly reference?)
```

Comment:
Same here. We forbid types named `scoped` in C# 11, right? This should be unambiguously a bad modifier, not a type name.